modules/core/arrow-validation/src/main/kotlin/arrow/validation/refinedTypes/string/NonEmptyString.kt

```diff
@@ -0,0 +1,39 @@
+package arrow.validation.refinedTypes.string
+
+import arrow.Kind
+import arrow.core.Either
+import arrow.core.EitherPartialOf
+import arrow.data.Nel
+import arrow.data.Validated
+import arrow.data.ValidatedPartialOf
+import arrow.extension
+import arrow.instances.either.applicativeError.applicativeError
+import arrow.instances.nonemptylist.semigroup.semigroup
+import arrow.instances.validated.applicativeError.applicativeError
+import arrow.typeclasses.ApplicativeError
+import arrow.validation.RefinedPredicateException
+import arrow.validation.Refinement
+
+interface NonEmptyString<F> : Refinement<F, String> {
```

Comment:
Can `NonEmpty<A>` be defined instead of specializing this constraint per type?

Reply:
Sure, we just have to define an empty value for type `A`